### ISSUE TITLE
SC2203 assumes * is always a globbing attempt, even though it can also be multiplication

### ISSUE THREAD

**Issue by josephcsible** [koalaman/shellcheck#1641]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2203
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

foo=3
bar=4

if [[ 2*$foo -gt $bar ]]; then
  echo yes
else
  echo no
fi

```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
 
Line 6:
if [[ 2*$foo -gt $bar ]]; then
      ^-- SC2203: Globs are ignored in [[ ]] except right of =/!=. Use a loop.

$ 
```

#### Here's what I wanted or expected to see:

Either no finding, or something that mentions multiplication. Since `*` is valid for multiplication, we shouldn't yell at the user for using it for globbing any time we see it.

**Comment by Gandalf-** [comment]
You could work around this and preserve the multiplication behavior with `\*`
```sh
foo=3
bar=4

if [[ 2\*$foo -gt $bar ]]; then
  echo yes
else
  echo no
fi
```
```
$ bash example.sh
yes
```

It would be easy enough to add a comment to this warning explaining "if you want multiplication here, we suggest `\*` for clarity".

**Comment by koalaman** [comment]
You're right, of course. You will no longer get this warning, and if you use `[ .. ]` you will get a new SC2255 suggesting that you use `$((..))` to expand your own expressions. Thanks!
